From brian@CSUA.Berkeley.EDU Thu Sep  9 20:25:32 1999
Return-Path: <brian@CSUA.Berkeley.EDU>
Received: from soda.CSUA.Berkeley.EDU (soda.CSUA.Berkeley.EDU [128.32.43.52])
	by hub.freebsd.org (Postfix) with ESMTP id 4DD7F152D2
	for <FreeBSD-gnats-submit@freebsd.org>; Thu,  9 Sep 1999 20:25:30 -0700 (PDT)
	(envelope-from brian@CSUA.Berkeley.EDU)
Received: from smarter.than.nu (ida-89-77.Reshall.Berkeley.EDU [169.229.89.77]) 
	by soda.CSUA.Berkeley.EDU (8.8.8/) via ESMTP id UAA17320
        for <FreeBSD-gnats-submit@freebsd.org>; Thu, 9 Sep 1999 20:24:55 -0700 (PDT)
        env-from (brian@CSUA.Berkeley.EDU)
Message-Id: <Pine.BSF.4.10.9909092019290.691-100000@smarter.than.nu>
Date: Thu, 9 Sep 1999 20:24:54 -0700 (PDT)
From: "Brian W. Buchanan" <brian@CSUA.Berkeley.EDU>
To: FreeBSD-gnats-submit@freebsd.org
Subject: [PATCH] Fix games/flightgear build under 3.3RC

>Number:         13674
>Category:       ports
>Synopsis:       Patch for games/flightgear for 3.3
>Confidential:   no
>Severity:       critical
>Priority:       high
>Responsible:    freebsd-ports
>State:          open
>Quarter:        
>Keywords:       
>Date-Required:  
>Class:          sw-bug
>Submitter-Id:   current-users
>Arrival-Date:   Thu Sep  9 20:30:00 PDT 1999
>Closed-Date:    
>Last-Modified:  
>Originator:     Brian Buchanan
>Release:        FreeBSD 4.0-CURRENT i386
>Organization:
>Environment:
>Description:

Satoshi's build logs show FlightGear failing under 3.3 due to some
implicit typecasting that the compiler has no problem with under 4.0, but
barfs at under 3.3.  A patch is attached.  I don't have any way to test
this under 3.3, nor do I know if anything later in the build process fails
under 3.3, so if someone could do a quick check and fix any other trivial
little things before the ports freeze, it would be appreciated.

>How-To-Repeat:
>Fix:

Add patches/patch-ag:

--- Simulator/GUI/gui.cxx.old	Thu Sep  9 20:11:55 1999
+++ Simulator/GUI/gui.cxx	Thu Sep  9 20:13:07 1999
@@ -149,7 +149,7 @@
     glutSetCursor(GLUT_CURSOR_INHERIT);
 #endif
 #if (GLUT_API_VERSION >= 4 || GLUT_XLIB_IMPLEMENTATION >= 9)
-    glutWarpPointer( glutGet(GLUT_SCREEN_WIDTH)/2, glutGet(GLUT_SCREEN_HEIGHT)/2);
+    glutWarpPointer( glutGet((GLenum)GLUT_SCREEN_WIDTH)/2, glutGet((GLenum)GLUT_SCREEN_HEIGHT)/2);
 #endif
 }
 
@@ -160,7 +160,7 @@
     glutSetCursor(GLUT_CURSOR_NONE);
 #else  // I guess this is what we want to do ??
 #if (GLUT_API_VERSION >= 4 || GLUT_XLIB_IMPLEMENTATION >= 9)
-    glutWarpPointer( glutGet(GLUT_SCREEN_WIDTH), glutGet(GLUT_SCREEN_HEIGHT));
+    glutWarpPointer( glutGet((GLenum)GLUT_SCREEN_WIDTH), glutGet((GLenum)GLUT_SCREEN_HEIGHT));
 #endif
 #endif
 }
@@ -195,7 +195,7 @@
     if( restore ) {
         glutSetCursor(cursor);
     } else {
-        cursor = glutGet( GLUT_WINDOW_CURSOR );
+        cursor = glutGet( (GLenum)GLUT_WINDOW_CURSOR );
 #ifdef WIN32
         TurnCursorOn();
 #endif
@@ -595,7 +595,7 @@
     // Install our fast fonts
     fntpath.append( "typewriter.txf" );
     guiFntHandle = new fntTexFont ;
-    guiFntHandle -> load ( fntpath.c_str() ) ;
+    guiFntHandle -> load ( (char *)fntpath.c_str() ) ;
     puFont GuiFont ( guiFntHandle, 15 ) ;
     puSetDefaultFonts( GuiFont, GuiFont ) ;
     guiFnt = puGetDefaultLabelFont();


>Release-Note:
>Audit-Trail:
>Unformatted:

